coremltools/converters/mil/frontend/torch/test/test_torch_ops.py

```diff
@@ -2419,6 +2419,49 @@ def forward(self, x):
 
         self.run_compare_torch(input_shape, TestModel(), backend=backend)
 
+    @pytest.mark.parametrize(
+        "input_shape, dim, mode, backend",
+        itertools.product([(2, 2), (1, 1)], [0, 1], ["min", "max"], backends)
+    )
+    def test_min_max_no_keepdim(self, input_shape, dim, mode, backend):
+        class TestModel(nn.Module):
```

Comment:
With this change we'll now have three separate copies of `TestModel`. We should really just have one copy. Let's make `TestModel` an inner class to `TestReduction`. Then each of these three test method can instantiate via `self.TestReduction`.

Reply:
Good idea, just a heads up there are a couple of unique edge cases that contribute to the end result of merging all these TestModels. I've listed them out here:

The following, outputs a 0 dim tensor with the minimum value of the entire array.
```
x = torch.rand((3,3))
torch.min(x)
```


As soon as a dim is provided torch returns a tuple of (values, indicies)
```
x = torch.rand((3,3))
torch.min(x,dim=0)
```

And finally, if provided with two tensors it'll do elementwise min and return a tensor of the same shape.
```
x = torch.rand((3,3))
y = torch.rand((3,3))
torch.min(x,y)
```

Given all those conditions, the code isn't exactly the most elegant, but it does what we want:
```
class TestReduction(TorchBaseTest):
  
    class TestModel(nn.Module):
        def __init__(self, mode, dim=None, keepdim=None):
            super().__init__()
            args = {"dim": dim, "keepdim": keepdim}

            self.op_args = {k:v for k, v in args.items() if v is not None}

            if mode == "min":
                self.op = torch.min
            elif mode == "max":
                self.op = torch.max
            else:
                raise ValueError("Unsupported mode: {mode}".format(mode=mode))
        
        def forward(self, x, y=None):

            if y is not None:
                return self.op(x, y)
            else:
                return self.op(x, **self.op_args)
```

I'll push this onto the branch, let me know if you have any other suggestions!